Application provided invalid, non monotonically increasing dts to muxer in stream 0: 466966 >= 466966

### Problem Description

I have 2 videos which fail to encode in HandBrake, using the same profile as the other 500 videos I've encoded, that are also very similar in terms of the source codecs, resolution, framerate, download source, etc.

The videos play completely fine in players like VLC and IINA.

I have also tried copying the data into a new mkv container with FFmpeg, which does copy the data without any errors, but even those new files have the same errors in HandBrake.

### Activity Log, Crash Log or any other details

```text
HandBrake Activity Log for Session: 2023-04-16T05:32:10Z
Handbrake Version: 1.6.1 (2023012200)
2017-01-08_En osaa tätä challangee - Minecraft： Skywars Challenge - #1 (16.4.2023).mp4
Preset: AV1 1080p RF 56 Preset 7
[08:32:10] macgui: fr.handbrake.HandBrakeXPCService work failed
[08:32:10] macgui: fr.handbrake.HandBrakeXPCService scanning specifically for title: 4
[08:32:10] CPU: Intel(R) Core(TM) i7-4960HQ CPU @ 2.60GHz
[08:32:10]  - Intel microarchitecture Haswell
[08:32:10]  - logical processor count: 8
[08:32:10] hb_scan: path=/Users/piipperi/Downloads/2017-01-08_En osaa tätä challangee - Minecraft： Skywars Challenge - #1.mkv, title_index=4
udfread ERROR: ECMA 167 Volume Recognition failed
disc.c:333: failed opening UDF image /Users/piipperi/Downloads/2017-01-08_En osaa tätä challangee - Minecraft： Skywars Challenge - #1.mkv
disc.c:437: error opening file BDMV/index.bdmv
disc.c:437: error opening file BDMV/BACKUP/index.bdmv
[08:32:10] bd: not a bd - trying as a stream/file instead
libdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdnav: vm: vm: failed to read VIDEO_TS.IFO
[08:32:10] dvd: not a dvd - trying as a stream/file instead
Input #0, matroska,webm, from '/Users/piipperi/Downloads/2017-01-08_En osaa tätä challangee - Minecraft： Skywars Challenge - #1.mkv':
  Metadata:
    MINOR_VERSION   : 0
    COMPATIBLE_BRANDS: iso6avc1mp41
    MAJOR_BRAND     : dash
    ENCODER         : Lavf59.27.100
  Duration: 00:13:37.62, start: -0.007000, bitrate: 4730 kb/s
  Stream #0:0: Video: h264 (High), yuv420p(tv, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 59.94 fps, 59.94 tbr, 1k tbn (default)
    Metadata:
      HANDLER_NAME    : ISO Media file produced by Google Inc. Created on: 11/07/2022.
      VENDOR_ID       : [0][0][0][0]
      DURATION        : 00:13:37.557000000
  Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
    Metadata:
      DURATION        : 00:13:37.621000000
[08:32:10] scan: decoding previews for title 4
[08:32:10] scan: audio 0x1: opus, rate=48000Hz, bitrate=1 English (Opus) (2.0 ch)
[08:32:10] scan: 10 previews, 1920x1080, 59.940 fps, autocrop = 0/0/96/96, aspect 16:9, PAR 1:1, color profile: 1-1-1, chroma location: left
[08:32:10] libhb: scan thread found 1 valid title(s)
[08:32:10] macgui: fr.handbrake.HandBrakeXPCService scan done
[08:32:10] Starting work at: Sun Apr 16 08:32:10 2023

[08:32:10] 1 job(s) to process
[08:32:10] macgui: fr.handbrake.HandBrakeXPCService started encoding 2017-01-08_En osaa tätä challangee - Minecraft： Skywars Challenge - #1 (16.4.2023).mp4
[08:32:10] macgui: fr.handbrake.HandBrakeXPCService with preset AV1 1080p RF 56 Preset 7
[08:32:10] Starting Task: Encoding Pass
[08:32:10] Skipping vfr filter
[08:32:10] Skipping crop/scale filter
[08:32:10] work: only 1 chapter, disabling chapter markers
[08:32:10] job configuration:
[08:32:10]  * source
[08:32:10]    + /Users/piipperi/Downloads/2017-01-08_En osaa tätä challangee - Minecraft： Skywars Challenge - #1.mkv
[08:32:10]    + title 4, chapter(s) 1 to 1
[08:32:10]    + container: matroska,webm
[08:32:10]    + data rate: 4730 kbps
[08:32:10]  * destination
[08:32:10]    + /Users/piipperi/Library/Containers/fr.handbrake.HandBrake/Data/Movies/2017-01-08_En osaa tätä challangee - Minecraft： Skywars Challenge - #1 (16.4.2023).mp4
[08:32:10]    + container: MPEG-4 (libavformat)
[08:32:10]      + align initial A/V stream timestamps
[08:32:10]  * video track
[08:32:10]    + decoder: h264 8-bit (yuv420p)
[08:32:10]    + filters
[08:32:10]      + Comb Detect (mode=0:spatial-metric=2:motion-thresh=2:spatial-thresh=3:filter-mode=1:block-thresh=80:block-width=16:block-height=16)
[08:32:10]      + Decomb (mode=39)
[08:32:10]    + Output geometry
[08:32:10]      + storage dimensions: 1920 x 1080
[08:32:10]      + pixel aspect ratio: 1 : 1
[08:32:10]      + display dimensions: 1920 x 1080
[08:32:10]    + encoder: AV1 (SVT)
[08:32:10]      + preset:  7
[08:32:10]      + profile: main
[08:32:10]      + level:   auto
[08:32:10]      + quality: 56.00 (RF)
[08:32:10]      + color profile: 1-1-1
[08:32:10]      + chroma location: left
[08:32:10]  * audio track 1
[08:32:10]    + decoder: English (Opus) (2.0 ch) (track 1, id 0x1)
[08:32:10]      + samplerate: 48000 Hz
[08:32:10]    + mixdown: Stereo
[08:32:10]    + encoder: AAC (Apple AudioToolbox)
[08:32:10]      + bitrate: 160 kbps, samplerate: 48000 Hz
[08:32:10] sync: expecting 49008 video frames
[08:32:10] encavcodecInit: AV1 encoder
[08:32:10] encavcodec: encoding at CRF 56.00
[08:32:10] encavcodec: encoding with stored aspect 1/1
Svt[info]: -------------------------------------------
Svt[info]: SVT [version]:	SVT-AV1 Encoder Lib v1.4.1
Svt[info]: SVT [build]  :	Apple LLVM 14.0.0 (clang-1400.0.29.202)	 64 bit
Svt[info]: LIB Build date: Jan 22 2023 22:41:52
Svt[info]: -------------------------------------------
Svt[info]: Number of logical cores available: 8
Svt[info]: Number of PPCS 54
Svt[info]: [asm level on system : up to avx2]
Svt[info]: [asm level selected : up to avx2]
Svt[info]: -------------------------------------------
Svt[info]: SVT [config]: main profile	tier (auto)	level (auto)
Svt[info]: SVT [config]: width / height / fps numerator / fps denominator 		: 1920 / 1080 / 60000 / 1001
Svt[info]: SVT [config]: bit-depth / color format 					: 8 / YUV420
Svt[info]: SVT [config]: preset / tune / pred struct 					: 7 / VQ / random access
Svt[info]: SVT [config]: gop size / mini-gop size / key-frame type 			: 604 / 16 / key frame
Svt[info]: SVT [config]: BRC mode / rate factor 					: CRF / 56 
Svt[info]: -------------------------------------------
libdvdread: Encrypted DVD support unavailable.
libdvdread: Can't open file VIDEO_TS.IFO.
[08:32:11] sync: first pts video is 0
[08:32:11] sync: "Chapter 1" (1) at frame 1 time 0
[08:32:11] sync: first pts audio 0x1 is 0
[mp4 @ 0x7f9fb9b0d740] Application provided invalid, non monotonically increasing dts to muxer in stream 0: 466966 >= 466966
[08:36:53] avformatMux: track 0, av_interleaved_write_frame failed with error 'Invalid argument'
[08:36:53] Last error repeated 1 times
[08:36:53] avformatMux: track 0, av_interleaved_write_frame failed with error 'Invalid argument'
[mp4 @ 0x7f9fb9b0d740] Application provided invalid, non monotonically increasing dts to muxer in stream 0: 466966 >= 466966
[08:36:53] work: average encoding speed for job is 19.959499 fps
[08:36:53] comb detect: heavy 7 | light 335 | uncombed 5278 | total 5620
[08:36:53] decomb: deinterlaced 7 | blended 335 | unfiltered 5269 | total 5611
[08:36:53] opus-decoder done: 4751 frames, 0 decoder errors
[08:36:53] h264-decoder done: 5677 frames, 0 decoder errors
[08:36:53] sync: got 5639 frames, 49008 expected
[08:36:53] sync: framerate min 59.940 fps, max 59.940 fps, avg 59.961 fps
Svt[error]: deinit called without sending EOS!
[08:36:55] mux: track 0, 5534 frames, 10233120 bytes, 870.96 kbps, fifo 8192
[08:36:55] mux: track 1, 4408 frames, 1641211 bytes, 139.69 kbps, fifo 8192
[08:36:55] Finished work at: Sun Apr 16 08:36:55 2023

[08:36:55] libhb: work result = 4
```


### What Operating System are you running?

macOS 12.6.5 (21G531)

### What version of HandBrake are you running?

1.6.1 (2023012200)

### Where did you download HandBrake from?

github

--- a/libhb/nvenc_common.c
+++ b/libhb/nvenc_common.c
@@ -289,8 +289,23 @@ int hb_nvdec_are_filters_supported(hb_list_t *filters)
     for (int i = 0; i < hb_list_count(filters); i++)
     {
         hb_filter_object_t *filter = hb_list_item(filters, i);
-        hb_deep_log( 2, "nvdec: %s isn't yet supported for CUDA video frames", filter->name);
-        ret = 0;
+
+        switch (filter->id)
+        {
+            case HB_FILTER_VFR:
+            {
+                // Mode 0 doesn't require access to the frame data
+                int mode = hb_dict_get_int(filter->settings, "mode");
+                if (mode == 0)
+                {
+                    break;
+                }
+            }
+            default:
+                hb_deep_log( 2, "nvdec: %s isn't yet supported for CUDA video frames", filter->name);
+                ret = 0;
+                break;
+        }
     }
 
     return ret;

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -4982,6 +4982,7 @@ int hb_qsv_sanitize_filter_list(hb_job_t *job)
     {
         int i = 0;
         int num_sw_filters = 0;
+        int num_hw_filters = 0;
         if (job->list_filter != NULL && hb_list_count(job->list_filter) > 0)
         {
             for (i = 0; i < hb_list_count(job->list_filter); i++)
@@ -4992,6 +4993,7 @@ int hb_qsv_sanitize_filter_list(hb_job_t *job)
                 {
                     // cropping and scaling always done via VPP filter
                     case HB_FILTER_CROP_SCALE:
+                        num_hw_filters++;
                         break;
                     case HB_FILTER_VFR:
                     {
@@ -5003,13 +5005,14 @@ int hb_qsv_sanitize_filter_list(hb_job_t *job)
                         }
                     }
                     default:
+                        // count only filters with access to frame data
                         num_sw_filters++;
                         break;
                 }
             }
         }
         job->qsv.ctx->num_sw_filters = num_sw_filters;
-        job->qsv.ctx->qsv_hw_filters_are_enabled = ((hb_list_count(job->list_filter) == 1) && hb_qsv_full_path_is_enabled(job)) ? 1 : 0;
+        job->qsv.ctx->qsv_hw_filters_are_enabled = ((num_hw_filters > 0) && hb_qsv_full_path_is_enabled(job)) ? 1 : 0;
         if (job->qsv.ctx->qsv_hw_filters_are_enabled)
         {
             job->qsv.ctx->hb_vpp_qsv_frames_ctx = av_mallocz(sizeof(HBQSVFramesContext));

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -4993,13 +4993,15 @@ int hb_qsv_sanitize_filter_list(hb_job_t *job)
                     // cropping and scaling always done via VPP filter
                     case HB_FILTER_CROP_SCALE:
                         break;
-
-                    case HB_FILTER_YADIF:
-                    case HB_FILTER_ROTATE:
-                    case HB_FILTER_RENDER_SUB:
-                    case HB_FILTER_AVFILTER:
-                        num_sw_filters++;
-                        break;
+                    case HB_FILTER_VFR:
+                    {
+                        // Mode 0 doesn't require access to the frame data
+                        int mode = hb_dict_get_int(filter->settings, "mode");
+                        if (mode == 0)
+                        {
+                            break;
+                        }
+                    }
                     default:
                         num_sw_filters++;
                         break;

--- a/libhb/vfr.c
+++ b/libhb/vfr.c
@@ -538,6 +538,11 @@ static void hb_vfr_close( hb_filter_object_t * filter )
         hb_log("vfr: %d frames output, %d dropped and %d duped for CFR/PFR",
                pv->count_frames, pv->drops, pv->dups );
     }
+    else
+    {
+        hb_log("vfr: %d frames output, %d dropped",
+               pv->count_frames, pv->drops);
+    }
 
     if( pv->job )
     {

--- a/libhb/work.c
+++ b/libhb/work.c
@@ -1387,27 +1387,7 @@ static void sanitize_filter_list(hb_job_t *job, hb_geometry_t src_geo)
         }
     }
 
-    int is_detel = 0;
-    hb_filter_object_t * filter = hb_filter_find(list, HB_FILTER_DETELECINE);
-    if (filter != NULL)
-    {
-        is_detel = 1;
-    }
-
-    filter = hb_filter_find(list, HB_FILTER_VFR);
-    if (filter != NULL)
-    {
-        int mode = hb_dict_get_int(filter->settings, "mode");
-        // "Same as source" FPS and no HB_FILTER_DETELECINE
-        if ( (mode == 0) && (is_detel == 0) )
-        {
-            hb_list_rem(list, filter);
-            hb_filter_close(&filter);
-            hb_log("Skipping vfr filter");
-        }
-    }
-
-    filter = hb_filter_find(list, HB_FILTER_CROP_SCALE);
+    hb_filter_object_t *filter = hb_filter_find(list, HB_FILTER_CROP_SCALE);
     if (filter != NULL)
     {
         hb_dict_t* settings = filter->settings;

